common/prometheus-server/templates/alerts/_prometheus.alerts.tpl

```diff
@@ -249,6 +249,40 @@ groups:
       summary: Prometheus scrapes pods multiple times
   {{- end }}
 
+  {{- if and (eq $root.Values.vpaUpdateMode "Auto") (not $root.Values.alerts.thanos.enabled) }}
+  - alert: PrometheusVpaMemoryExceeded
+    expr: |
+      vpa_butler_vpa_container_recommendation_excess{verticalpodautoscaler=~"{{ include "prometheus.fullName" . }}",resource="memory"} / 1024 / 1024 / 1024 > 0
+    for: 15m
+    labels:
+      service: {{ include "alertServiceLabelOrDefault" "metrics" }}
+      support_group: {{ include "alertSupportGroupOrDefault" "observability" }}
+      severity: info
+      meta: Prometheus VPA for `{{`{{ $labels.verticalpodautoscaler }}`}}` in `{{`{{ $labels.namespace }}`}}` is recommending more memory.
+    annotations:
+      description: |
+        `{{`{{ $labels.verticalpodautoscaler }}`}}` in `{{`{{ $labels.cluster }}/{{ $labels.namespace }}`}}` needs more `{{`{{ $labels.resource }}`}}`. It is overutilized by `{{`{{ $value }}`}}` GiB.
+        It is hitting the VPA maxAllowed boundary and is not ensured to run properly at its current place. Consider upgrading the VPA maxAllowed
+        memory value if the host memory size permits.
+      summary: Prometheus needs more memory.
+
+  - alert: PrometheusVpaCPUExceeded
+    expr: |
+      vpa_butler_vpa_container_recommendation_excess{verticalpodautoscaler=~"{{ include "prometheus.fullName" . }}",resource="cpu"} > 0
+    for: 15m
+    labels:
+      service: {{ include "alertServiceLabelOrDefault" "metrics" }}
+      support_group: {{ include "alertSupportGroupOrDefault" "observability" }}
+      severity: info
+      meta: Prometheus VPA for `{{`{{ $labels.verticalpodautoscaler }}`}}` in `{{`{{ $labels.namespace }}`}}` is recommending more CPUs.
+    annotations:
+      description: |
+        `{{`{{ $labels.verticalpodautoscaler }}`}}` in `{{`{{ $labels.cluster }}/{{ $labels.namespace }}`}}` needs more `{{`{{ $labels.resource }}`}}`. It is overutilized by `{{`{{ $value }}`}}` cores.
+        It is hitting the VPA maxAllowed boundary and is not ensured to run properly at its current place. Consider upgrading the VPA maxAllowed
+        CPU core value if the host memory size permits.
+      summary: Prometheus needs more CPU.
+  {{- end }}
+
```

Comment:
Why is this needed? It would only work in the kubernetes case.
Since we leverage the prometheus scoped thanos rule, it can be removed, no?

Reply:
The entire common chart is built on the assumption that prometheus-kubernetes is the default. If this is not the case, we need to forward the alerts to the Thanos rule. This switch is here:
```
{{- if and (eq $root.Values.vpaUpdateMode "Auto") (not $root.Values.alerts.thanos.enabled) }}
```
ans the values:
https://github.com/sapcc/helm-charts/blob/ed1f5d9dc8226113b55a6e65936d996c22266987/common/prometheus-server/values.yaml#L417-L420

Reply:
To label them with the specified `thanos-ruler`, there is another PrometheusRule in that [file](https://github.com/sapcc/helm-charts/blob/ed1f5d9dc8226113b55a6e65936d996c22266987/common/prometheus-server/templates/alerts/prometheus-alerts.yaml).

Reply:
Thanks for elaborating. Suggesting two comments to remember future me of this nexus.